Rubberduck.VBEEditor/Events/AutoCompleteEventArgs.cs

```diff
@@ -1,57 +1,64 @@
-                    OldCode = module.GetLines(selection.PreviousLine);
-                }
-                else
-                {
-                    OldCode = module.GetLines(selection);
-                }
+                Keys = e.Key;
             }
+            else
+            {
+                Character = e.Character;
+            }
+            CodeModule = module;
+            CurrentSelection = module.GetQualifiedSelection().Value.Selection;
+            CurrentLine = module.GetLines(CurrentSelection);
         }
 
         /// <summary>
-        /// The CodePane wrapper for the module being edited.
+        /// <c>true</c> if the character has been handled, i.e. written to the code pane.
+        /// Set to <c>true</c> to swallow the character and prevent the WM message from reaching the code pane.
+        /// </summary>
+        public bool Handled { get; set; }
+
+        /// <summary>
+        /// The CodeModule wrapper for the module being edited.
         /// </summary>
-        public ICodePane CodePane { get; }
+        public ICodeModule CodeModule { get; }
 
         /// <summary>
-        /// Indicates whether the line of code held in <see cref="OldCode"/> is committed or not.
+        /// <c>true</c> if the event is originating from a <c>WM_CHAR</c> message.
+        /// <c>false</c> if the event is originating from a <c>WM_KEYDOWN</c> message.
         /// </summary>
         /// <remarks>
-        /// If the line is committed, <see cref="OldCode"/> is located on the line that precedes the current selection in the <see cref="CodePane"/>.
+        /// Inline completion is handled on WM_CHAR; deletions and block completion on WM_KEYDOWN.
         /// </remarks>
-        public bool IsCommitted { get; }
-
+        public bool IsCharacter => Keys == default;
         /// <summary>
-        /// The content hash for the module before autocompletion. Used to prevent misfiring autocompletes.
+        /// The character whose key was pressed. Undefined value if <see cref="Keys"/> isn't `<see cref="Keys.None"/>.
         /// </summary>
-        public string ContentHash { get; }
-
+        public char Character { get; }
         /// <summary>
-        /// If not committed, the entire current line of code. If committed, the line of code immediately preceding the current selection.
+        /// The actionnable key that was pressed. Value is <see cref="Keys.None"/> when <see cref="IsCharacter"/> is <c>true</c>.
         /// </summary>
-        public string OldCode { get; }
+        public Keys Keys { get; }
 
         /// <summary>
-        /// The autocompleted line of code, assigned by the autocomplete implementation. Used for caching, to prevent misfiring autocompletes.
-        /// If autocomplete works off committed input, this should match the <see cref="OldCode"/>.
+        /// The current location of the caret.
+        /// </summary>
+        public Selection CurrentSelection { get; }
```

Comment:
if it's the location of the caret why not name it exactly that? `CaretLocation` seems semantically much more accurate than `CurrentSelection`.

Reply:
But then it would only be true when the selection is a single character?